packages/controls/src/controls/associated-types.ts

```diff
@@ -0,0 +1,31 @@
+import { ControlDefinition } from './definition'
+
+export type ControlType<D> =
+  D extends ControlDefinition<infer ControlType, any, any, any, any, any>
+    ? ControlType
+    : never
```

Comment:
Funny story with these ones: originally they were formulated as

```typescript

export type AssociatedTypes<D> =
  D extends ControlDefinition<
    infer ControlType,
    infer Config,
    infer DataType,
    infer ValueType,
    infer ResolvedValueType,
    infer InstanceType
  >
    ? {
        ControlType: ControlType
        Config: Config
        DataType: DataType
        ValueType: ValueType
        ResolvedValueType: ResolvedValueType
        InstanceType: InstanceType
      }
    : never

export type ControlType<D> = AssociatedTypes<D>['ControlType']
export type DataType<D> = AssociatedTypes<D>['DataType']
...
```
... which seems like a less repetitive but otherwise equivalent way to infer these types. However, it turned out that in some situations the above is NOT equivalent to the current formulation. I'll produce a sandbox reproduce for this at some point.